src/profile/helpers/updateMutationVariables.ts

```diff
@@ -28,139 +36,146 @@ type AddressInputs = {
   removeAddresses?: (string | null)[] | null | undefined;
 };
 
-export const getAddress = (addresses: Address[], profile?: MyProfileQuery) => {
-  const profileAddresses: Address[] = [
-    profile?.myProfile?.primaryAddress as Address,
-    ...getAddressesFromNode(profile),
-  ];
+type PhoneInputs = {
+  addPhones: CreatePhoneInput[];
+  updatePhones: UpdatePhoneInput[];
+  removePhones?: (string | null)[] | null | undefined;
+};
 
-  const updateAddresses: UpdateAddressInput[] = addresses
-    .filter(address => {
-      const profileAddress = profileAddresses.find(
-        profileAddress => profileAddress?.id === address.id
-      );
+const getPrimaryValue = (primary: Primary, profile?: MyProfileQuery) => {
+  const primaryValue = profile?.myProfile && profile.myProfile[primary];
+  return primaryValue || { id: '' };
+};
 
-      return address.id && !isEqual(address, profileAddress);
-    })
-    .map(address => {
+const getEmptyObject = (primary: Primary) => {
```

Comment:
It's not obvious what this is for, maybe a comment explaining the purpose could be helpful!

Reply:
I'll add a comment. My imagination was running on fumes when I was naming these 😄